### PR TITLE
Issue 14760 - Add clearContentLength for avoiding segmentation fault

### PR DESCRIPTION
**Pull request by karronoli**
clearContentLength method make content length option clear as same as contentLength property for cleaing the extra option.

It is done to test the changes by persistent connection.

### CODE REVIEW

**Comment by DmitryOlshansky** [comment]
Please add `Fix issue 14760` to the commit message, then it's going to close the ticket on this pull request merge.

**Comment by DmitryOlshansky** [comment]
Well, now just squash these 2 commits into one and it should be fine. Thanks!

**Comment by karronoli** [comment]
Sorry. I have squashed the commits.
Regards,

**Comment by karronoli** [comment]
Sorry. I have squashed the commits.
Regards,

**Comment by quickfur** [comment]
LGTM

File: std/net/curl.d [hunk 1]
```diff
@@ -2807,6 +2808,35 @@ struct HTTP
         }
     }
 
+    /// Clear ContentLength depending on method.
```
Comment (DmitryOlshansky) [inline std/net/curl.d]: This is internal function as far as I can tell. Then do *not* use DDoc comment just `//` is enough.

Comment (karronoli) [inline std/net/curl.d]: I see. I will reflect it.

**Comment by burner** [comment]
LGTM

**Comment by DmitryOlshansky** [comment]
Auto-merge toggled on

**Comment by MartinNowak** [comment]
This fix causes a bug on 32-bit systems. If the next request doesn't set data, an uninitialized content length will be used.
I get this output for the test case.
```
* Connected to d-lang.appspot.com (64.233.166.141) port 80 (#0)
> POST /testUrl2 HTTP/1.1
User-Agent: Phobos-std.net.curl/2.069 (libcurl/7.37.0)
Host: d-lang.appspot.com
Accept: */*
Content-Length: 11
Content-Type: application/x-www-form-urlencoded

< HTTP/1.1 200 OK
< Cache-Control: no-cache
< Content-Type: application/octet-stream
< Expires: Fri, 01 Jan 1990 00:00:00 GMT
< Date: Thu, 20 Aug 2015 10:08:01 GMT
* Server Google Frontend is not blacklisted
< Server: Google Frontend
< Content-Length: 11
< 
* Connection #0 to host d-lang.appspot.com left intact
* Found bundle for host d-lang.appspot.com: 0x92fb218
* Re-using existing connection! (#0) with host d-lang.appspot.com
* Connected to d-lang.appspot.com (64.233.166.141) port 80 (#0)
> TRACE /testUrl1 HTTP/1.1
User-Agent: Phobos-std.net.curl/2.069 (libcurl/7.37.0)
Host: d-lang.appspot.com
Accept: */*
Content-Length: 582261328974970880
Content-Type: application/x-www-form-urlencoded
Expect: 100-continue

* HTTP 1.0, assume close after body
< HTTP/1.0 400 Bad Request
< Content-Type: text/html; charset=UTF-8
< Content-Length: 1419
< Date: Thu, 20 Aug 2015 10:08:01 GMT
* Server GFE/2.0 is not blacklisted
< Server: GFE/2.0
< 
* Closing connection 0
```

The best solution for now is to explicitly set the content length to 0.
For a full reuse of a curl handle we need to make [curl_easy_reset](http://curl.haxx.se/libcurl/c/curl_easy_reset.html) available, see http://curl.haxx.se/mail/curlphp-2011-04/0002.html.

**Comment by MartinNowak** [comment]
See https://github.com/D-Programming-Language/phobos/pull/3567.

**Pull request merged**